app/javascript/article-form/components/Preview.jsx

```diff
@@ -80,17 +81,33 @@ const previewResponsePropTypes = PropTypes.shape({
 });
 
 export const Preview = ({
+  previewLoading,
   previewResponse,
   articleState,
   errors,
   markdownLintErrors,
 }) => {
   useEffect(() => {
-    if (previewResponse.processed_html.includes('twitter-timeline')) {
+    if (previewResponse?.processed_html?.includes('twitter-timeline')) {
       attachTwitterTimelineScript();
     }
   }, [previewResponse]);
 
+  if (previewLoading) {
+    const coverImage = articleState.mainImage;
+    let loadingPreview;
+    if (coverImage === null) {
+      loadingPreview = <LoadingPreview version="default" />;
+    } else {
+      loadingPreview = <LoadingPreview version="cover" />;
+    }
```

Comment:
```suggestion
    const loadingPreview =  <LoadingPreview version={coverImage === null ? "default" : "cover"} />
```
just a suggestion / not blocking - it could even be in-lined on line 106

Reply:
Applied this change.